### DISCUSSION TITLE
if_completed_before - 30 minutes ?

### DISCUSSION THREAD

**Discussion by KlavsKlavsen** [inducer/relate#923] [Q&A]
I am trying to figure out, if I can use the if_completed_before example under grading to get every question in the flow (I have 40 questions) - render 4 points per question IF within the first 30 minutes of the session start - and only 1 point per question if AFTER 30 minutes from start.
Is that supported ?

This seems to set the Grading rule - exactly like that for the all flows in this repo - is that correct? https://github.com/inducer/relate-sample/blob/main/yaml-macros.yml#L30
but where can I read what is supported in this line:
```
if_completed_before: end_week {{ week }}
``` 
?
would it support like:
```
if_completed_before: 30m
```
? and is that counting from session start or can it ONLY look at "time now" ?

**Comment by inducer** [comment]
This is a good example of why I'd like to be able to write flow rules in actual code, cf. https://github.com/inducer/starlark-pyo3/.

In the meantime, I think your best bet is to add another flow rule conditional, maybe `if_completed_within_minutes`? Just analogously extend this code:

https://github.com/inducer/relate/blob/2186f96bfb1bb45d498c162cf8d824e9a84f6c4a/course/utils.py#L492-L513
